frontend/app/services/user-service.server.ts

```diff
@@ -0,0 +1,55 @@
+import moize from 'moize';
+
+import { getAuditService } from '~/services/audit-service.server';
+import { getInstrumentationService } from '~/services/instrumentation-service.server';
+import { getEnv } from '~/utils/env.server';
+import { getLogger } from '~/utils/logging.server';
+
+const log = getLogger('user-service.server');
+
+export const getUserService = moize(createUserService, { onCacheAdd: () => log.info('Creating new user service') });
+
+function createUserService() {
+  const { CDCP_API_BASE_URI } = getEnv();
+
+  async function createUserWithAttributesAndEmail(email: string, userId: string) {
+    const auditService = getAuditService();
+    const instrumentationService = getInstrumentationService();
+    auditService.audit('user-service.create', { userId });
+
+    const newUser = {
+      email: email,
+      userAttributes: [
+        {
+          name: 'RAOIDC_USER_ID',
+          value: userId,
+        },
+      ],
+    };
+
+    const url = new URL(`${CDCP_API_BASE_URI}/api/v1/users`);
+    const response = await fetch(url, {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify(newUser),
+    });
+
+    instrumentationService.countHttpStatus('http.client.cdcp-api.users.posts', response.status);
+
+    if (!response.ok) {
+      log.error('%j', {
+        message: 'Failed creating user',
+        status: response.status,
+        statusText: response.statusText,
+        url: url,
+        responseBody: await response.text(),
+      });
+      throw new Error(`Failed to create user. Status: ${response.status}, Status Text: ${response.statusText}`);
+    }
+    return response.status;
```

Comment:
I think this function can be a void return type. I don't see any scenarios where the routes using this service function will need to use the status.

Reply:
From my other comment:

> `userService.createUser()` should return a representation of the user, including the passed-in data, but also the new user id that the API creates.

If the API doesn't currently return the user, then it needs to be fixed.

Reply:
should return the user now